### PR TITLE
(feat) AI Assistant 

### PR DESCRIPTION
**Pull request by abj10**
# **(feat) AI Chat Assistant**

### **Screenshots**

Capture d’écran 2024-12-06 à 02 42 53
Capture d’écran 2024-12-06 à 02 45 11

---

### **What?**

This PR introduces initial integration of an AI Chat Assistant. It includes repositories, navigation updates, and UI components to support the new feature.

---

#### 1. **AI Assistant Repository**
- **`AiAssistantRepository.kt`:** Handles AI-related data management and interactions.

#### 2. **AI Assistant Screen**
- **`AiAssistantScreen.kt `:** Handles the UI of the AI Assistant.

#### 3. **UI Enhancements**
- **`DashboardScreen.kt`:** Updated to accommodate AI assistant functionalities.
- **`NavigationActions.kt`:** Added routes for AI Assistant-related screens.

#### 4. **Backend Preparations**
- Modified `MainActivity.kt` to initialize AI-related dependencies.
- Updated `Widget.kt` with assistant-specific attributes.

#### 5. **Testing Framework**
- **`assistant/AiAssistantScreenTest`:** Added tests for the UI.

---

### **Why?**
These updates enable:
- Seamless integration of AI-powered features in the Eduverse ecosystem.
- Enhanced user experience through intelligent assistant capabilities.

---

### **How?**
- Implemented repository patterns for managing AI Assistant data.
- Updated navigation and UI to reflect the addition of AI functionalities.
- Created a test structure to ensure feature reliability.

---

### **Type of Change**
- [x] New Feature (AI Assistant integration)
- [x] Enhancement (navigation and UI updates)

---

### **Checklist**
- [x] Code adheres to project standards.
- [x] Verified navigation flows and repository functionality.
- [x] Integrated UI components with backend services.
- [x] Added UI and Unit tests.

---

#### **New Files**
- `app/src/main/java/com/github/se/eduverse/repository/AiAssistantRepository.kt`
- `app/src/main/java/com/github/se/eduverse/ui/assistant/AiAssistantScreen.kt`
- `app/src/test/java/com/github/se/eduverse/repository/AssistantTest.kt`
- `app/src/androidTest/java/com/github/se/eduverse/ui/assistant/`

#### **Core Files**
- `app/src/main/java/com/github/se/eduverse/MainActivity.kt`
- `app/src/main/java/com/github/se/eduverse/model/Widget.kt`
- `app/src/main/java/com/github/se/eduverse/ui/dashboard/DashboardScreen.kt`
- `app/src/main/java/com/github/se/eduverse/ui/navigation/NavigationActions.kt`

### CODE REVIEW

**Review by AKallala — CHANGES_REQUESTED** [review]
Well done on implementing this new feature, it adds substantial value to the app. However I pointed out some changes that need to be done. 
First, the changes made to DashboardScreen need to be rolled back, as they overwrite modifications introduced in #247. This is also the reason the CI is currently failing. Additionally, I suggest creating a view model for this feature to better adhere to MVVM principles and improve the separation of concerns. I’ve also provided several other suggestions to enhance code quality.

File: app/src/main/java/com/github/se/eduverse/ui/dashboard/DashboardScreen.kt
[the remark this answers is not in the document]
Comment (abj10) [reply app/src/main/java/com/github/se/eduverse/ui/dashboard/DashboardScreen.kt:1]: Yes indeed, there was a problem while resolving conflicts, I did a revert to last commit before merge to solve the problem

**Comment by abj10** [comment]
> You're still discarding changes merged to main branch.

I didn't discard anything, it's an issue generated by the reverts. Indeed previous merge main into this branch caused problems

**Comment by AKallala** [comment]
This PR has been recreated in a new PR #283 because of the merge issues

**Pull request closed without merging**